grpcbp/prometheus.go

```diff
@@ -100,3 +102,16 @@ var (
 		Help: "The number of in-flight requests",
 	}, clientActiveRequestsLabels)
 )
+
+// serviceAndMethodSlug splits the UnaryServerInfo.FullMethod and returns
+// the package.service part separate from the method part.
+// ref: https://pkg.go.dev/google.golang.org/grpc#UnaryServerInfo
+func serviceAndMethodSlug(fullMethod string) (service string, method string) {
+	split := strings.SplitN(fullMethod, "/", 3)
+	if len(split) < 2 {
+		return "", ""
+	}
+	method = split[len(split)-1]
+	service = strings.Join(split[:len(split)-1], "")
```

Comment:
I... really don't know what I should expect from the full method in a grpc call. the example they gave will have `split[0]` being the empty string and `split[1]` being the service fqn, so this join works. but is there a scenario that `split[0]` will be non-empty? and if that happens, do we want to just join that against `split[1]`?

can someone link to some grpc doc to explain what to expect here? the ref linked above only provided one example.

Reply:
[gRPC's http/2 interface uses /service/method as the path -- see PROTOCOL-HTTP2 and search for :path](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md)